service/src/main/java/bio/terra/workspace/service/workspace/AwsCloudContextService.java

```diff
@@ -140,19 +162,35 @@ public AwsCloudContext getRequiredAwsCloudContext(UUID workspaceUuid) {
   public @NotNull Environment discoverEnvironment()
       throws IllegalArgumentException, InternalLogicException {
     try {
-      Assert.notNull(this.environmentDiscovery, "environmentDiscovery not configured");
+      if (this.environmentDiscovery == null) {
+        throw new InvalidApplicationConfigException("AWS environmentDiscovery not initialized");
+      }
+      return environmentDiscovery.discoverEnvironment();
 
-      Environment environment = environmentDiscovery.discoverEnvironment();
-      Assert.notNull(environment, "environment null");
-      Assert.notNull(environment.getMetadata(), "environment.metadata null");
-      Assert.notEmpty(environment.getSupportedRegions(), "environment.landingZones empty");
-
-      return environment;
-
-    } catch (IllegalArgumentException e) {
-      throw new InvalidApplicationConfigException("AWS configuration error: " + e.getMessage());
     } catch (IOException e) {
       throw new InternalLogicException("AWS discover environment error", e);
     }
   }
+
+  /**
+   * Return the landing zone to for the given cloud context's region
+   *
+   * @param awsCloudContext AWS cloud context
+   * @param region AWS region
+   * @return AWS landing zone, if supported for the Cloud context region
+   * @throws StaleConfigurationException StaleConfigurationException
+   */
+  public Optional<LandingZone> getLandingZone(
+      @NotNull AwsCloudContext awsCloudContext, @NotNull Region region) {
+    Environment environment = discoverEnvironment();
+
+    AwsCloudContext awsCloudContextFromEnv = getCloudContext(environment);
+    if (!awsCloudContext.equals(awsCloudContextFromEnv)) {
```

Comment:
I think that this is a good thing to check, but object equality feels a little coarse.

Maybe we should just check against some of the more "invariant" fields, specifically the major version number and the AWS account number feel like things that would be a problem if they changed, whereas the "alias" fields and the org are not really meaningful to the rest of the WSM now.

Reply:
Checking the tenant and env definitely are an overkill atm. 
since there was already a equals method, leveraged it. Plus, all the fields ought to be the same anyway, so decided to not write code to check individual fields